make/compiler/Makefile.sanitizers

```diff
@@ -0,0 +1,40 @@
+# Copyright 2020 Hewlett Packard Enterprise Development LP
+# Copyright 2004-2019 Cray Inc.
+# Other additional copyright holders may be indicated within.
+#
+# The entirety of this work is licensed under the Apache License,
+# Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.
+#
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Note that for ASAN gcc has leak sanitizer on by default, which isn't that
+# helpful ATM. Disable with `export ASAN_OPTIONS=detect_leaks=0`
+
+ifneq ($(CHPL_MAKE_SANITIZERS), none)
+  # Configuration checks
+  ifneq ($(CHPL_MAKE_TASKS), fifo)
+    $(error CHPL_TASKS=fifo is required for sanitizers)
+  endif
+  ifneq ($(CHPL_MAKE_TARGET_MEM), cstdlib)
+    $(error CHPL_MEM=cstdlib is required for sanitizers)
+  endif
+
+  # enable sanitizers
+  SANITIZER_CFLAGS += -fsanitize=$(CHPL_MAKE_SANITIZERS)
+  SANITIZER_LDFLAGS += -fsanitize=$(CHPL_MAKE_SANITIZERS)
+
+  # If debugging with optimizations, lower optimization level and enable
+  # frame pointers for better stack traces
+  ifeq ($(DEBUG), 1)
+    SANITIZER_OPT_CFLAGS += -fno-omit-frame-pointer -O1
```

Comment:
Do we throw `-O` or `-O3` or whatever **always** in the backend compilation? Otherwise we maybe adding unintentional optimization flag when `DEBUG==1`

Reply:
This gets appended to `OPT_FLAGS`, which is only thrown at build time with `OPTIMIZE=1` (the default) or to the backend with `-O` / `--fast`. This will not add optimizations flags if optimizations are not enabled (that was the problem in the early versions you saw and that was causing the `var-tracking-assignments` warnings)